feat(wallet): allow setting custom nonce in transaction panel

onyb opened the pull request:
Resolves https://github.com/brave/brave-browser/issues/19705.

@bbondy It seems the `txData.baseData.nonce` field in `TransactionInfo` is set as an empty string by default. This prevents us from displaying the nonce that's used for creating the transaction.

cc: @jamesmudgett to review the UX.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Demo

https://user-images.githubusercontent.com/3684187/149326198-a1d01f1a-a080-4623-ab09-833aca3188d8.mov

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/f/wallet/custom-nonce/300abfd6bf06bc1dd51da549e280f25d0e9b3b02-a301e748c92bd2bd2d9e9f77300e713d63928a7e/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/f/wallet/custom-nonce/ca5c75147d0af97e7598ff9c9723c50df618ef58-e1b5f40d4b1b7929af14e009ffe89d52e7909b4e/storybook/index.html)

onyb commented:
Small update to the UX based on feedback from James. Changed the placeholder text, and added an info message.

<img width="300" src="https://user-images.githubusercontent.com/3684187/149557325-0f14fbd4-c59e-4f65-b0b6-876a5212d0c3.png">

onyb commented:
Commit https://github.com/brave/brave-core/pull/11840/commits/2dbc0d029c162df5e90a75426aed2a15d9029851 LGTM. ✔️ 
Tested OK ✔️.
